### PR TITLE
Checking for conflicts for created or updated keys

### PR DESCRIPTION
**Pull request by itwillwork**
There is currently no conflict check for created or updated keys, so you may lose the key.

Testing example:
image

### CODE REVIEW

File: src/models/Keyset.ts [hunk 1]
```diff
@@ -83,7 +101,6 @@ export class Keyset implements KeysetBase {
       }
 
       this._keyset.keyset.status[name][lang] = langs?.[lang]?.allowedStatus;
-      AllowedStatuses.GENERATED;
```
Comment (itwillwork) [inline src/models/Keyset.ts:86]: useless command

File: src/models/Keyset.ts [hunk 2]
```diff
@@ -60,8 +78,8 @@ export class Keyset implements KeysetBase {
     return this.value;
   };
 
-  updateKeys = (butchPayload) => {
-    for (const payload of butchPayload) {
+  updateKeys = (batchPayload) => {
```
Comment (itwillwork) [inline src/models/Keyset.ts:81]: fix typo

File: src/models/Keyset.ts [hunk 3]
```diff
@@ -29,7 +29,25 @@ export class Keyset implements KeysetBase {
...
+      const currentValue = (langs[lang] as LangPayload)?.value;
+      const prevValue = this._keyset[lang][name];
+      if (prevValue && currentValue !== prevValue) {
+        hasConflict = true;
+      }
+    }
+
+    return hasConflict;
+  }
+
```
Comment (itwillwork) [inline src/models/Keyset.ts:45]: Checking for conflicts while writing to a file

File: src/transform.ts [hunk 4]
```diff
@@ -93,11 +93,24 @@ export default async function (file: FileInfo, api: API, options: Options) {
...
+          currentKeyData && (
+              currentKeyData?.ru !== newKeyData?.ru ||
+              currentKeyData?.en !== newKeyData?.en
+          )
+      ) {
+        throw new Error(`Conflict for keyName: "${keyName}" check usage!`);
+      }
+
+      // store keysets data in temporal object
+      data[keyset as string].keyData[keyName as string] = newKeyData;
```
Comment (itwillwork) [inline src/transform.ts:113]: checking for conflicts during key accumulation

**Review by IsaevAlexandr — APPROVED** [review]

**Pull request merged**